mig/shared/install.py

```diff
@@ -521,29 +524,49 @@ def generate_confs(
                 _GENERATE_CONFS_NOFORWARD_KEYS}
 
     # expand any directory path specific as "auto" relative to CWD
-    thecwd = _getcwd()
 
     if source == keyword_auto:
         # use the templates from this copy of the code tree
         template_dir = os.path.join(MIG_BASE, "mig/install")
     else:
         # construct a path using the supplied value made absolute
-        template_dir = abspath(source, start=thecwd)
+        template_dir = abspath(source, start=generateconfs_output_path)
 
     if destination == keyword_auto:
         # write output into a confs folder within the CWD
-        destination = os.path.join(thecwd, 'confs')
+        destination = os.path.join(generateconfs_output_path, 'confs')
+    elif os.path.isabs(destination):
+        # take the caller at face-value and do not change the path
+        pass
     else:
         # construct a path from the supplied value made absolute
-        destination = abspath(destination, start=thecwd)
+        destination = abspath(destination, start=generateconfs_output_path)
 
     # finalize destination paths up-front
     destination_link = destination
     destination_dir = "%s%s" % (destination, destination_suffix)
 
+    if mig_code == keyword_auto:
+        expanded['mig_code'] = os.path.join(MIG_BASE, 'mig')
+        mig_code = expanded['mig_code']
+
+    # expand any directory path specified "auto" relative to output location
+
+    if mig_certs == keyword_auto:
+        expanded['mig_certs'] = os.path.join(
```

Comment:
We don't actually want the `mig_certs` value to be inside the generated confs dir. It's a _runtime_ path set in MiGserver.conf etc. and should generally remain relative to `MIG_BASE` unless specifically provided. If you look carefully PR75 had that, too :)